app/Filament/Resources/Budgeting/IncomeResource/Actions/QuickBudgetAction.php

```diff
@@ -44,12 +48,31 @@ protected function setUp(): void
                                 ->where('income_id', $this->getRecord()->getKey())
                         )
                     ),
+                Repeater::make('history')
+                    ->visible(fn (Income $income) => $income->is_fluctuating)
+                    ->collapsible()
+                    ->label(__('filament-forms::components.repeater.label.income.history'))
+                    ->schema([
+                        TextInput::make('description')
+                            ->label(__('filament-forms::components.text_input.label.description.name'))
+                            ->required()
+                            ->maxLength(255),
+                        MoneyInput::make('amount')
+                            ->label(__('filament-forms::components.text_input.label.money.name'))
+                            ->required(),
+                        DatePicker::make('revenue_at')
+                            ->label(__('filament-forms::components.text_input.label.income.history_date'))
+                            ->required()
+                            ->maxDate(now()),
+                    ]),
             ])
             ->modalHeading(fn (?Income $record): string => __('budgetly::actions.income.modal_heading.create').': '.$record?->name)
             ->action(function (array $data, Income $record, QuickBudgetAction $action, Form $form, array $arguments): void {
                 $data['income_id'] = $record->id;
 
-                $record->budgets()->create($data);
+                $budget = $record->budgets()->createOrFirst($data);
+
+                $budget->histories()->createMany($data['history']);
```

Comment:
i think we need an conditional tho?  what if the quick budget action is not fluctuated? i don't think we still need to insert the histories? wdyt?

Reply:
oh yeah, i forgot validate on this action